### DISCUSSION TITLE
Syntax highlighting

### DISCUSSION THREAD

**Discussion by allanleal** [thuliteio/doks#87] [Q&A]
I find Doks one of the most elegant website templates for documentation. I'm wondering if Doks supports syntax highlighting (e.g., Python, C++, etc.). I could not find any code blocks with syntax highlighting in https://getdoks.org/.

**Comment by h-enk** [comment]
Thanks!

Doks supports the Hugo [syntax highlighting](https://gohugo.io/content-management/syntax-highlighting). However I "switched it off" per default because I was not satisfied with the [Chroma styles](https://xyproto.github.io/splash/docs/) available (to match with the Doks look and feel). My idea was to implement the [GitHub light syntax theme](https://github.com/primer/github-syntax-light) (best match with the Doks look and feel — in my opinion) and was in the process of converting it to a Chroma style (I was halfway and realized a 100% match is not feasible — so I stalled it). Also, the [GitHub Chroma](https://xyproto.github.io/splash/docs/longer/github.html) style is not what I had in mind (not really a match with the Primer style — I think).

That having said, you can enable syntax highlighting by uncommenting line 17 of `./assets/scss/app.scss`:

```scss
// @import "components/syntax";
```

You would then get my half-half GitHub style. However, you could also generate an other style by following the [Hugo instructions](https://gohugo.io/content-management/syntax-highlighting#generate-syntax-highlighter-css) and copy that in `./assets/scss/components/_syntax.scss`

Note: for production, make sure to add to the `...whitelister([` section of  `purgecss` in `./config/postcss.config.js`:

```js
          './assets/scss/components/_syntax.scss',
```

Apart from Hugo syntax highlighting, you could also implement a npm package like e.g. [prismjs](https://www.npmjs.com/package/prismjs).

Hope this helps.

**Reply by aghwotu** [reply]
Hello there @h-enk . Your theme is absolutely gorgeous. I am using it to document some internal API and it has been fun working with it. I have the same issue and I have uncommented line 17 but commented it back and pulled in prismjs with npm instead but it is not having any effect on the code blocks. Please can you help me? Thank you again for a great theme.

**Reply by h-enk** [reply]
Hello @aghwotu, nice to hear that you like Doks!

Would you mind sharing your repo? Then I can help you better.

**Reply by aghwotu** [reply]
Hello @h-enk. Sorry for the late response. It is a private repo. Is it okay if I add you as a collaborator? Thank you again.

**Reply by h-enk** [reply]
Hello @aghwotu, that's OK

**Reply by aghwotu** [reply]
Hello @h-enk I have added you to the repo 😀

**Reply by h-enk** [reply]
Hey @aghwotu, I've created a PR 😀

**Reply by aghwotu** [reply]
@h-enk I have merged it. It is superb! Thank you! You're awesome.

**Comment by allanleal** [comment]
Hi @h-enk - I'll also try out your solution later. I would recommend you however to have a switch/option in Doks to more easily enable this feature.

**Reply by vivekragunathan** [reply]
Yeah, it would be highly convenient if the code theme itself (at least for the well-known ones) can be specified in the config file

```
codeTheme = dracula
```

Something like that.

**Reply by h-enk** [reply]
Yeah, I think so too: https://github.com/h-enk/doks/issues/699
